pennylane_lightning/lightning_kokkos/_state_vector.py

```diff
@@ -26,6 +52,454 @@ class LightningKokkosStateVector:  # pylint: disable=too-few-public-methods
         device_name(string): state vector device name. Options: ["lightning.kokkos"]
+        kokkos_args(InitializationSettings): binding for Kokkos::InitializationSettings
+            (threading parameters).
+        sync(bool): immediately sync with host-sv after applying operations
+
     """
 
-    pass  # pylint: disable=unnecessary-pass
+    def __init__(
+        self,
+        num_wires,
+        dtype=np.complex128,
+        device_name="lightning.kokkos",
+        kokkos_args=None,
+        sync=True,
+    ):  # pylint: disable=too-many-arguments
+        self._num_wires = num_wires
+        self._wires = Wires(range(num_wires))
+        self._dtype = dtype
+
+        self._kokkos_config = {}
+
+        if dtype not in [np.complex64, np.complex128]:  # pragma: no cover
+            raise TypeError(f"Unsupported complex type: {dtype}")
+
+        if device_name != "lightning.kokkos":
+            raise DeviceError(f'The device name "{device_name}" is not a valid option.')
+
+        self._device_name = device_name
+
+        # self._qubit_state = self._state_dtype()(self._num_wires)
+        if kokkos_args is None:
+            self._kokkos_state = self._state_dtype()(self.num_wires)
+        elif isinstance(kokkos_args, InitializationSettings):
+            self._kokkos_state = self._state_dtype()(self.num_wires, kokkos_args)
+        else:
+            type0 = type(InitializationSettings())
+            raise TypeError(
+                f"Argument kokkos_args must be of type {type0} but it is of {type(kokkos_args)}."
+            )
+        self._sync = sync
+
+        if not self._kokkos_config:
+            self._kokkos_config = self._kokkos_configuration()
+
+    @property
+    def dtype(self):
+        """Returns the state vector data type."""
+        return self._dtype
+
+    @property
+    def device_name(self):
+        """Returns the state vector device name."""
+        return self._device_name
+
+    @property
+    def wires(self):
+        """All wires that can be addressed on this device"""
+        return self._wires
+
+    @property
+    def num_wires(self):
+        """Number of wires addressed on this device"""
+        return self._num_wires
+
+    @property
+    def state_vector(self):
+        """Returns a handle to the state vector."""
+        return self._kokkos_state
+
+    @property
+    def state(self):
+        """Copy the state vector data from the device to the host.
+
+        A state vector Numpy array is explicitly allocated on the host to store and return
+        the data.
+
+        **Example**
+
+        >>> dev = qml.device('lightning.kokkos', wires=1)
+        >>> dev.apply([qml.PauliX(wires=[0])])
+        >>> print(dev.state)
+        [0.+0.j 1.+0.j]
+        """
+        state = np.zeros(2**self._num_wires, dtype=self.dtype)
+        state = self._asarray(state, dtype=self.dtype)
+        self.sync_d2h(state)
+        return state
```

Comment:
@vincentmr  Here is where `_array` is used

Reply:
Seems like `state` is already of type `self.dtype`. Why do we need to pass it through `_asarray`?

Reply:
Yes, I think the same. I did not change it because is a `@staticmethod` then I did not know if it would be necessary in other places. However, I run a test without the function and everything looks fine. Therefore, I will remove it in the next commit. 
Thank you. :+1:

Reply:
I believe it is outdated, but I may be wrong. Let's try to removing it at some point.

Reply:
Done, I have already deleted it. If it is necessary in the future we will know.